vms/avm/wallet_service_backend.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package avm
+
+import (
+	"context"
+
+	"golang.org/x/exp/maps"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/linked"
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/ava-labs/avalanchego/vms/avm/txs"
+	"github.com/ava-labs/avalanchego/vms/components/avax"
+	"github.com/ava-labs/avalanchego/wallet/chain/x/builder"
+)
+
+var _ txBuilderBackend = (*walletServiceBackend)(nil)
+
+func NewWalletServiceBackend(vm *VM) *walletServiceBackend {
+	backendCtx := &builder.Context{
+		NetworkID:        vm.ctx.NetworkID,
+		BlockchainID:     vm.ctx.XChainID,
+		AVAXAssetID:      vm.feeAssetID,
+		BaseTxFee:        vm.Config.TxFee,
+		CreateAssetTxFee: vm.Config.CreateAssetTxFee,
+	}
+
+	return &walletServiceBackend{
+		ctx:        backendCtx,
+		vm:         vm,
+		pendingTxs: linked.NewHashmap[ids.ID, *txs.Tx](),
+		utxos:      make([]*avax.UTXO, 0),
+	}
+}
+
+type walletServiceBackend struct {
+	ctx        *builder.Context
+	vm         *VM
+	pendingTxs *linked.Hashmap[ids.ID, *txs.Tx]
+	utxos      []*avax.UTXO
+
+	addrs set.Set[ids.ShortID]
+}
+
+func (b *walletServiceBackend) Context() *builder.Context {
+	return b.ctx
+}
+
+func (b *walletServiceBackend) ResetAddresses(addrs set.Set[ids.ShortID]) {
+	b.addrs = addrs
+}
```

Comment:
I think we should be able to remove this method + `addrs` field in this struct if we make the users a parameter in the function args instead

Reply:
Dropped ResetAddresses by introducing an adapter among the AVMBuilderBackend and the wallet Builder